lib/Monad_WP/OptionMonad.thy

```diff
@@ -62,6 +62,30 @@ lemma opt_map_upd_Some:
 
 lemmas opt_map_upd[simp] = opt_map_upd_None opt_map_upd_Some
 
+lemma opt_map_Some_comp:
+  "f ||> h o g = f ||> g ||> h"
```

Comment:
Hard to tell without opening Isabelle, but this and the four following after the next one all look like they would be nice to have as `[simp]`.

Reply:
I added this one to the simp set as well, but it seems that this needs to be in the opposite direction. I will fix it in my next batch. It affects my WIP only, I am pretty sure :)

Reply:
The test worked fine after the merge, so all good.